### PR TITLE
docs: pipeline retrospective + Sprint 3 PR-2 handoff

### PR DESCRIPTION
**Pull request by komleff**
## Summary

Два docs-only документа после Sprint 3 PR-1 cycle (PR #203 merged 2026-05-10):

### docs/ideas/2026-05-10-pipeline-improvements-from-sprint-3-pr-1.md

Discussion material (НЕ план) с 7 предложениями pipeline improvement из PR-1 retrospective. Каждый пункт: Observation / Proposed change / Why / связанные артефакты + memory references.

**7 пунктов:**
1. Adversarial pass = Pass 2 для Critical tier (по умолчанию)
2. External review pre-flight health check
3. Smoke test matrix (short / long / restart-resilience)
4. PM verdict authority при convergence detection
5. Mode C explicit hard rule (не external review-pass)
6. Smoke instructions: git status warning ДО destructive команды
7. PM consultation rule reinforcement (anti-pattern examples)

**Cumulative impact estimate:** 50% reduction review-cycle wall-clock (5h → 2.5h) при сохранении merge quality.

**Recommended ordering** для будущего применения: #6 → #7 → #1 → #4 → #3 → #5 → #2 (по cost/benefit).

### docs/handoffs/2026-05-10-sprint-3-pr-2-handoff.md

Handoff для следующего PM-агента, который начнёт Sprint 3 PR-2 (T0.3 Prediction & Smoothing, `U2-91p`) в новой сессии. Содержит:

- Промпт активации
- Текущий state репозитория + branch для работы
- Scope PR-2 (port от 2D testbed PredictionEngine + Hermite smoothing + adaptive render-delay buffer)
- Архитектурные инварианты (Plane Universe, server-authoritative, c'=3000, 30 Hz)
- 18 deferred Beads из PR-1 cycle с context
- Lessons learned из PR-1 cycle (КРИТИЧНО прочитать)
- 2D testbed референсы для port'а
- Anti-patterns из retrospective

Replaces необходимость читать ~5h transcript Sprint 3 PR-1 cycle для PR-2 pickup.

## Tier

**Tier: Light** — docs-only, not affects code paths. Без external review.

## Test plan

- [ ] Frontmatter YAML валиден в обоих файлах
- [ ] Internal links не сломаны (Memory entries в `~/.claude/projects/.../memory/` referenced through file names)
- [ ] PR-2 PM-сессия может pickup через handoff без re-reading transcript

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: docs/handoffs/2026-05-10-sprint-3-pr-2-handoff.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+1. **Adversarial pass — главный win.** Используй adversarial framing для второго internal pass (Critical tier 2-pass requirement) с focus'ом на race conditions / edge cases. Не делать 5 standard passes как в PR-1.
+2. **Smoke test matrix:** short / long-window 60s+ / restart-resilience. Reconciliation jitter под high RTT может быть скрытой за коротким smoke. Documented в `Test plan` PR description.
+3. **PM convergence triage authority:** при iter raised только P2 — defer to Beads с rationale, **не reverse в fix-now**. ADR 3.24 streak считается "0 NEW Critical/High".
+4. **Mode C ≠ external review:** если Codex CLI недоступен → escalate operator для Mode D, не silent fallback в Claude vs Claude.
+5. **PM не консультирует на pipeline-mandated шагах:** fix-cycle / verify / pass2 — auto, не AskUserQuestion.
+6. **git status warning ДО destructive команды:** в operator instructions Шаг 0 — git status check.
+
+Memory entries (auto-loaded в новую сессию): см. `MEMORY.md` index. Ключевые для PR-2:
+- `feedback_pm_no_consultation_on_pipeline_steps.md`
```
[the remark this answers is not in the document]
Comment (komleff) [reply docs/handoffs/2026-05-10-sprint-3-pr-2-handoff.md]: Valid finding. Memory entries (`MEMORY.md`, `feedback_*.md`) лежат в Claude Code machine-local state (`~/.claude/projects/d--GitHub-u2/memory/`), не в репо — fresh машина / fresh session не имеет к ним доступа.

Inline'ил все 7 правил в handoff section "Lessons learned" с формальной структурой **Когда / Что / Почему** для self-contained pickup без зависимости от machine-local state. Пример (Правило 1):

> **Когда:** второй internal review pass для Critical tier.
>
> **Что:** не запускай вторым standard 4-аспектным pass; запускай adversarial-framed (промпт inline ниже).
>
> **Почему:** В Sprint 3 PR-1 cycle 5 standard passes APPROVED, потом adversarial Pass 6 (Mode C) нашёл 2 P0 race conditions. Если adversarial был Pass 2 — sane patch появился бы за 2 iter вместо 6+.

Также добавлены language tags на 4 fenced code blocks (`text` / `markdown`) per MD040.

Commit: `fff47ed`.
